core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3106,23 +3116,27 @@ private void setAttributeInternal(RpcContext rpcContext, LockedInodePath inodePa
   @Override
   public void scheduleAsyncPersistence(AlluxioURI path)
       throws AlluxioException, UnavailableException {
-    // We retry an async persist request until ufs permits the operation
     try (RpcContext rpcContext = createRpcContext();
         LockedInodePath inodePath = mInodeTree.lockFullInodePath(path, LockPattern.WRITE_INODE)) {
-      if (!inodePath.getInodeFile().isCompleted()) {
-        throw new InvalidPathException(
-            "Cannot persist an incomplete Alluxio file: " + inodePath.getUri());
-      }
-      mInodeTree.updateInode(rpcContext, UpdateInodeEntry.newBuilder()
-          .setId(inodePath.getInode().getId())
-          .setPersistenceState(PersistenceState.TO_BE_PERSISTED.name())
-          .build());
-      mPersistRequests.put(inodePath.getInode().getId(), new alluxio.time.ExponentialTimer(
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_INTERVAL_MS),
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_INTERVAL_MS),
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_WAIT_TIME_MS),
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_TOTAL_WAIT_TIME_MS)));
+      scheduleAsyncPersistenceInternal(rpcContext, inodePath.getInodeFile());
+    }
+  }
+
+  private void scheduleAsyncPersistenceInternal(RpcContext rpcContext, InodeFile inode)
```

Comment:
@gpang Due to how rename works, we don't have a LockedInodePath that can be used, I would prefer to make the internal method take LockedInodePath, do you have any suggestions? Another alternative is not to have the internal method and have slightly modified implementation at both call sites.

Reply:
Is the issue that we don't have a locked inode path from the call site? It looks like we have the full src path, but only the dst dir, so no path exists for the full dst path?

Reply:
I updated it to repeat some of the logic but avoid breaking any invariants around the method signatures.